src/pages/in/embed/_id/list.vue

```diff
@@ -83,6 +83,25 @@ import {
 
 import { MEDIUM_REGEX } from '~/constant';
 
+/* Although firebase fn should not have access to sensitive internal res,
+  and this isolation behaviour should be used as main defense,
+  still implement basic filter on url to avoid SSRF
+*/
+const checkValidDomainNotIP = (url) => {
+  const match = url.match(/^(?:http(?:s)?:\/\/)?([a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)+)(?::\d+)?/);
+  if (!match || !match[1]) return false;
+  const parts = match[1].split('.');
+  const isIP = (parts.length === 4 && parts.every((part) => {
+    try {
+      const partNum = Number(part);
+      return partNum >= 0 && partNum <= 255;
```

Comment:
If `part` is something like `0x1` or `1e1`, the predicate will accidentally return true
Should add additional regular expression to first check that the string is purely number:
```
if (!/^\d{1,3}$/.match(part)) return false;
```

(In case you want pure regular expression black magic: `/^(?:\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])$/`)